libc/math/common.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Math tests common part.
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "common.h"
+
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <float.h>
+
+double test_getRandomLog(double min, double max)
+{
+	if (min == 0.0) {
+		min = DBL_MIN;
+	}
+	double a = log(min);
+	double b = log(max);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+double test_getRandomLogSubnrm()
+{
+	double a = log(5.0e-324);
+	double b = log(2.0e-308);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+void test_setup(void)
+{
+	double num = M_PI;
+	double one = 1.0;
+	double zero = 0.0;
+
+	test_common.maxPowTwoPi = 0;
+	test_common.maxPowTwo = 0;
+	test_common.dblSignif = 0;
+
+	/* Initialize seed for test */
+	srand(time(0));
+
+	/* Initialize maxPowTwoPi value */
+	do {
+		test_common.maxPowTwoPi++;
+		num *= 2.0;
+	} while (num - M_PI != num);
+
+	num = 1.0;
+	/* Initialize maxPowTwo value */
+	do {
+		test_common.maxPowTwo++;
+		num *= 2.0;
+	} while (num + 0.5 != num);
+
+	num = 1.0;
+	/* Initialize digSignif value */
+	do {
+		test_common.dblSignif++;
+		num *= 2.0;
+	} while ((((num + one) - num) - one) == zero);
+}
+
+double test_checkResult(double f, double g)
+{
+	double digSignif, digLost, diff;
+
+	if (g != 0.0) {
+		diff = fabs((f - g) / g);
```

Comment:
Hm, in fabs test you use test_checkResult function, in test_checkResult you use fabs. I don't looking that carefully but please be sure that this doesn't cause that test will be malfunctioning. If It's asserted somehow, please place comment about it how it's achieved.